docs/src/main/asciidoc/spanner.adoc

```diff
@@ -0,0 +1,149 @@
+The `SpannerDialect` produces SQL, DML, and DDL statements for most common entity types and relationships using standard Hibernate and Java Persistence annotations.
+
+Please see the following sections for important details and differences due to Cloud Spanner's features and data model.
+
+=== Quick Set-Up
+
+Maven coordinates for the dialect:
+
+[source,xml]
+----
+<dependency>
+  <groupId>com.google.cloud</groupId>
+  <artifactId>google-cloud-spanner-hibernate-dialect</artifactId>
+  <version> TBD </version>
+</dependency>
+----
+
+Maven coordinates for the official Cloud Spanner Driver:
+
+[source,xml]
+----
+<dependency>
+  <groupId>com.google.cloud</groupId>
+  <artifactId>google-cloud-spanner</artifactId>
+  <version> TBD </version>
+</dependency>
+----
+
+Configuring the `SpannerDialect` and a Cloud Spanner Driver class is typical of all Hibernate dialects in the `hibernate.properties` file:
+
+----
+hibernate.dialect=com.google.cloud.spanner.hibernate.SpannerDialect
+hibernate.connection.driver_class=com.google.cloud.spanner.jdbc.JdbcDriver
+hibernate.connection.url=jdbc:cloudspanner:/projects/{INSERT_PROJECT_ID}/instances/{INSERT_INSTANCE_ID}/databases/{INSERT_DATABASE_ID}
+----
+
+The service account JSON credentials file's location should be in the `GOOGLE_APPLICATION_CREDENTIALS` environment variable.
+
+The dialect and driver are compatible with all values (`create`, `create-drop`, and `update`) of the `hibernate.hbm2ddl.auto` setting.
+
+
+=== Spanner Dialect
+
+The `SpannerDialect` supports most of the standard Hibernate and Java Persistence annotations but there are important differences in features because of differences in https://cloud.google.com/spanner/docs/schema-and-data-model[Cloud Spanner's data model] from traditional SQL databases.
+
+
+==== Relationships
+
+The dialect supports all of the standard relational annotations:
+
+- `@OneToOne`
+- `@OneToMany`
+- `@ManyToOne`
+- `@ManyToMany`
+
+These can be used via `@JoinTable` or `@JoinColumn`.
+However, because Cloud Spanner does not support Foreign Keys, foreign-key-columns are just regular columns in Cloud Spanner.
+
+NOTE: The lack of foreign keys also means database-side cascading deletes are not supported via the `@OnDelete(action = OnDeleteAction.CASCADE)` annotation because there is no `ON DELETE CASCADE` constraint in Cloud Spanner DDL.
+However, Hibernate-side cascading operations such as `@ManyToOne(cascade = {CascadeType.ALL})` are supported.
+
+
+==== Constraints
+
+Cloud Spanner does not support constraints.
+As a result, `SpannerDialect` does not currently support any constraints such as `FOREIGN KEY`, `UNIQUE`, or `ON DELETE CASCADE`.
+
+NOTE: Hibernate doesn't directly rely on the existence of constraints to perform its operations and leaves the enforcement relationship links to the database.
+
+
+==== Collections
+
+`SpannerDialect` supports collection properties annotated with `@ElementCollection`.
```

Comment:
Presumably anything that is not mentioned as unsupported, would be supported by the dialect.